### PR TITLE
fix: sorting error reporting audits backend filtering

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
https://2u-internal.atlassian.net/browse/ENT-6532

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: integrated_channels/api/v1/logs/views.py [hunk 1]
```diff
@@ -155,6 +170,16 @@ def get_queryset(self):
             # The serialized `sync_last_attempted_at` value is the representation of both the `modified` and `created` values
             if "sync_last_attempted_at" in content_sort_by[0]:
                 order = "-" if "-" in content_sort_by[0] else ""
-                content_sort_by = (f"{order}modified", f"{order}created")
+                # Arbitrarily old date to make sure null dates are considered lower in the `Greatest` annotation than any existing or to be
+                # created datetime in the audits
+                long_ago = datetime(year=1995, month=6, day=18)
+                return ThisLearnerClass.objects.annotate(
+                    oldest_date=Greatest(
```
[the remark this answers is not in the document]
Comment (alex-sheehan-edx) [reply integrated_channels/api/v1/logs/views.py:177]: lol good catch 👍 it should be newest not oldest.

File: integrated_channels/api/v1/logs/views.py [hunk 2]
```diff
@@ -104,7 +108,18 @@ def get_queryset(self):
...
+                long_ago = datetime(year=1995, month=6, day=18)
+                return ContentMetadataItemTransmission.objects.annotate(
+                    newest_date=Greatest(
+                        Coalesce('remote_updated_at', Cast(long_ago, DateTimeField()), output_field=DateTimeField()),
+                        Coalesce('remote_deleted_at', Cast(long_ago, DateTimeField()), output_field=DateTimeField()),
+                        Coalesce('remote_created_at', Cast(long_ago, DateTimeField()), output_field=DateTimeField()),
+                    )
+                ).filter(
+                    **content_metadata_transmission_audit_filter
+                ).order_by(f'{order}newest_date')
```
Comment (marlonkeating) [inline integrated_channels/api/v1/logs/views.py:122]: nit: I'm thinking we could encapsulate most of the annotation/sort logic in a helper method
```suggestion
                return SortByDateAggregation(objects=ContentMetadataItemTransmission.objects.filter(**content_metadata_transmission_audit_filter), date_fields=['remote_updated_at','remote_deleted_at','remote_created_at'])
```

**Review by marlonkeating — APPROVED** [review]
Looks good!

**Review by johnnagro — APPROVED** [review]
⛵

**Pull request merged**